import "item.proto";
import "common.proto";
//import "arena_shop.proto";
import "stage.proto";
import "line_up.proto";
import "pvp_rank.proto";
import "item_response.proto";
import "runt.proto";

//1241,1243,1245,1246,1248,1247, 1249
message positionRequest
{
	required int32 position = 1; //探索位置
}

//矿点信息
message mineData
{
	required int32 position = 1;//地图位置, 0玩家自己的初始矿
	required int32 type = 2; //矿点类型 1玩家占领的野怪矿，2野外矿，3神秘商人，4巨龙宝箱，5副本
	required int32 status = 3;//矿点状态1生产中，2可收获，3已枯竭，4空闲，5已领取，6副本已进入
	optional string nickname = 4;//昵称
	optional int32 last_time = 5;//生产结束时间点
	optional int32 gen_time = 6;//生产时间
	optional bool is_guild = 7;
	optional bool is_friend = 8;
	optional int32 fight_power = 9;
	optional int32 seek_help = 10;
	optional int32 mine_id = 11;
}

//更新矿点信息1240返回
message mineUpdate
{
	required int32 reset_today = 1; //今日已重置次数
	required int32 reset_free = 2; //可免费次数
	required int32 reset_count = 3; //可刷新总次数
	repeated mineData mine = 4;//所有位置信息
}

//1241返回
message searchResponse
{
	required CommonResponse res = 1;
	required int32 position = 2;
	required mineData mine = 3;//矿点信息
}

//1242请求 地图重置
message resetMap
{
	required int32 free = 1;//重置地图，1免费，2自费
}

//1242返回
message resetResponse
{
	required CommonResponse res = 1;
	required GameResourcesResponse consume = 2;
	required mineUpdate mine = 3;//所有位置信息
	required int32 free = 4;//重置地图，1免费，2自费
}

//符文石数据
message stoneData
{
	required int32 stone_id = 1;
	required int32 stone_num = 2;
}

//查看矿点详细信息返回1243
message mineDetail
{
	required int32 position = 1;//位置
	required CommonResponse res = 2;//通用返回
	required mineData mine = 3;
	required int32 limit = 4;
	repeated stoneData normal = 5;//符文石数据
	repeated stoneData lucky = 6;//符文石数据
	required int32 increase = 7;//增产剩余时间，单位：秒
	optional LineUpResponse lineup = 8;//驻守英雄信息
	optional int32 genUnit = 9;//单位产出:个数/小时
	optional float rate = 10;//增产百分比
	optional int32 incrcost = 11;//增产花费
	optional int32 stage_id = 12; //关卡id
	optional int32 guard_time = 13;//保护时间
	optional int32 accelerate_times = 14;
	optional int32 seek_help = 15;
}

//1245返回
message drawStones
{
	required int32 position = 1;
	required CommonResponse res = 2;//通用返回
	optional GameResourcesResponse normal = 3;
	optional GameResourcesResponse lucky = 4;
}

//神秘商人兑换1247请求，查看矿点兑换列表
//message shopRequest
//{
//	required int32 position = 1; //位置
//}

//密境神秘商人商品状态
message shopInfo
{
	required int32 shop_id = 1;//兑换商品id
	required int32 status = 2;//商品状态0进行中，1已兑换
}

//神秘商人兑换请求1247的返回
message shopStatus
{
	required int32 position = 1; //位置
	repeated shopInfo shop = 2; //本矿点出售商品信息
	required CommonResponse res = 3;
}

//密境神秘商人兑换请求1248请求
message exchangeRequest
{
	required int32 position = 1;//位置
	required int32 shop_id = 2;//兑换的商品id
}
//1248返回
message exchangeResponse
{
	required int32 position = 1;//位置
	required int32 shop_id = 2;//兑换商品id
    required CommonResponse res = 3;
    required GameResourcesResponse consume = 4;
    required GameResourcesResponse gain = 5;
}
//1249 返回使用item_response.ItemUseResponse
message boxReward
{
	required int32 position = 1;//位置
	required ItemUseResponse data = 2; 
}

//增产返回1250
message IncreaseResponse
{
	required int32 position = 1;//位置
	required int32 last_time = 2;//增产剩余时间，单位：秒
	required GameResourcesResponse consume = 3;
	required CommonResponse res = 4;
}


// 秘境相关, 驻守矿请求command:1244
// 返回CommonResponse
message MineGuardRequest
{
    required int32 pos = 1;                    // 矿在地图上的位置
    repeated MineLineUpSlot line_up_slots = 2; // 阵容
    optional int32 best_skill_id = 3;          // 无双id
	repeated int32 lineup = 4;      // 队形
}
//秘境相关，卡牌位
message MineLineUpSlot
{
    required int32 slot_no = 1;                    // 阵容位置
    required int32 hero_no  = 2;                   // 武将编号
    repeated MineEquipmentSlot equipment_slots = 3; // 装备
}
//秘境相关，装备
message MineEquipmentSlot
{
    required int32 slot_no = 1;      // 装备位置
    required string equipment_id = 2; // 装备ID
}

//秘境相关，战斗开始, command:1253
message MineBattleRequest
{
	required int32 pos = 1;          // 矿位置
	repeated int32 lineup = 2;      // 队形
	optional int32 unparalleled = 3; // 无双技能ID
	optional int32 fid = 4;          // 好友ID
	optional int32 hold = 5; //是否占领 0:抢夺，1抢夺并占领
}

// 秘境相关，战斗响应，command:1253
message MineBattleResponse{
	required CommonResponse res = 1;
	repeated BattleUnit red = 2;         // 红方数据 自己
	repeated BattleUnit blue = 3;    // 对方数据
	optional int32 red_best_skill_id = 4;       // 无双
	optional int32 red_best_skill_level = 5; // 无双
	optional int32 blue_best_skill_id = 6;       // 无双
	optional int32 blue_best_skill_level = 7; // 无双
	repeated int32 awake_no = 8;        //
	optional int32 seed1= 9;
	optional int32 seed2= 10;
	optional GameResourcesResponse gain = 11;
	optional bool fight_result = 12; // pvp
	optional int32 hold = 13; //是否占领 0:抢夺，1抢夺并占领
}

//返回 StageStartResponse

//秘境相关，pvp战斗开始, command:1251
//请求 PvpFightRequest
//返回 PvpFightResponse

//秘境相关， 战斗结算, command: 1252
//请求
message MineSettleRequest
{
    required int32 pos = 1;  //矿位置
    optional bool result = 2; //战斗结果
	repeated StepInfo steps = 3; // 战斗步骤
}
//返回 CommonResponse 或者自己定义协议

message MineSpeed{
}

message MineAccelerateRequest{
    required int32 pos = 1;  //矿位置
}

message MineAccelerateResponse{
	required CommonResponse res = 1;
	required GameResourcesResponse consume = 2;
}
